protostar/commands/test/starkware/cheatable_execute_entry_point.py

```diff
@@ -0,0 +1,422 @@
+            call_type=CallType.CALL,
+            class_hash=None,
+            contract_address=self.contract_address,
+            code_address=None,
+            entry_point_selector=self.entry_point_selector,
+            entry_point_type=self.entry_point_type,
+            calldata=self.calldata,
+            caller_address=self.caller_address,
+        )
+
+        return await call.execute(
+            state=state,
+            general_config=general_config,
+            tx_execution_context=self.get_execution_context(
+                n_steps=general_config.invoke_tx_max_n_steps
+            ),
+        )
+
+
+class CheatableExecuteEntryPoint(ExecuteEntryPoint):
+    def _run(
+        self,
+        state: CarriedState,
+        general_config: StarknetGeneralConfig,
+        loop: asyncio.AbstractEventLoop,
+        tx_execution_context: TransactionExecutionContext,
+    ) -> Tuple[CairoFunctionRunner, syscall_utils.BusinessLogicSysCallHandler]:
+        """
+        Runs the selected entry point with the given calldata in the code of the contract deployed
+        at self.code_address.
+        The execution is done in the context (e.g., storage) of the contract at
+        self.contract_address.
+        Returns the corresponding CairoFunctionRunner and BusinessLogicSysCallHandler in order to
+        retrieve the execution information.
+        """
+        # Prepare input for Cairo function runner.
+        class_hash = self._get_class_hash(state=state)
+        contract_class = state.get_contract_class(class_hash=class_hash)
+        contract_class.validate()
+
+        entry_point = self._get_selected_entry_point(
+            contract_class=contract_class, state=state
+        )
+
+        # Run the specified contract entry point with given calldata.
+        with wrap_with_stark_exception(code=StarknetErrorCode.SECURITY_ERROR):
+            runner = CairoFunctionRunner(program=contract_class.program, layout="all")
+        os_context = os_utils.prepare_os_context(runner=runner)
+
+        # Extract pre-fetched contract state from carried state.
+        pre_run_contract_carried_state = state.contract_states[self.contract_address]
+        contract_state = pre_run_contract_carried_state.state
+        contract_state.assert_initialized(contract_address=self.contract_address)
+
+        starknet_storage = BusinessLogicStarknetStorage(
+            commitment_tree=contract_state.storage_commitment_tree,
+            ffc=state.ffc,
+            # Pass a copy of the carried storage updates (instead of a reference) - note that
+            # pending_modifications may be modified during the run as a result of an internal call.
+            pending_modifications=dict(pre_run_contract_carried_state.storage_updates),
+            loop=loop,
+        )
+
+        initial_syscall_ptr = cast(
+            RelocatableValue, os_context[starknet_abi.SYSCALL_PTR_OFFSET]
+        )
+
+        # --- PROTOSTAR MODIFICATIONS ---
```

Comment:
Maybe it'd be better to use `BEGIN` and `END` with explicit commentary explaining what exactly was changed.

```python
# BEGIN: replace BusinessLogicSysCallHandler with CheatableSysCallHandler

# END
```